src/luascript.cpp

```diff
@@ -12175,6 +12171,133 @@ int LuaScriptInterface::luaPartySetSharedExperience(lua_State* L)
 	return 1;
 }
 
+// Spells
+int LuaScriptInterface::luaSpellCreate(lua_State* L)
+{
+	// Spell(words, name or id)
+	InstantSpell* spell;
+	if (isString(L, 2)) {
```

Comment:
isString uses lua_isstring which will return true for numbers and strings, you should first check if it's number using isNumber and then check with isString. I don't even think this is working for numbers as that else if will never get triggered. 

```cpp
if (isNumber(L, 2)) {
	spell = g_spells->getInstantSpellById(getNumber<uint32_t>(L, 2));
} else if (isString(L, 2)) {
	const std::string& stringArgument = getString(L, 2);
	spell = g_spells->getInstantSpellByName(stringArgument);
	if (!spell) {
		spell = g_spells->getInstantSpell(stringArgument); 
	}
}
```